homeassistant/components/sensor/mold_indicator.py

```diff
@@ -41,66 +44,125 @@
 })
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
     """Set up MoldIndicator sensor."""
     name = config.get(CONF_NAME, DEFAULT_NAME)
     indoor_temp_sensor = config.get(CONF_INDOOR_TEMP)
     outdoor_temp_sensor = config.get(CONF_OUTDOOR_TEMP)
     indoor_humidity_sensor = config.get(CONF_INDOOR_HUMIDITY)
     calib_factor = config.get(CONF_CALIBRATION_FACTOR)
 
-    add_entities([MoldIndicator(
+    async_add_entities([MoldIndicator(
         hass, name, indoor_temp_sensor, outdoor_temp_sensor,
-        indoor_humidity_sensor, calib_factor)], True)
+        indoor_humidity_sensor, calib_factor)], False)
+
+    return True
 
 
 class MoldIndicator(Entity):
     """Represents a MoldIndication sensor."""
 
-    def __init__(self, hass, name, indoor_temp_sensor, outdoor_temp_sensor,
-                 indoor_humidity_sensor, calib_factor):
+    def __init__(self, hass, name, indoor_temp_sensor,
```

Comment:
Pass in `is_metric` instead of `hass`. `hass` will be set on the entity when it has been added to home assistant.

Reply:
Most often it's a bad pattern yes. But `hass` will not be set until the entity has been added. So we can't use `self.hass` before that. It's ok to use `self.hass` in `update` and `async_added_to_hass` but not in `__init__`.